From a57c83496735c6babc3d1c2f631577c5ef7efca0 Mon Sep 17 00:00:00 2001
From: xxxbugxxxx <xxxbugxxxx@users.noreply.github.com>
Date: Fri, 15 Apr 2016 21:56:28 -0700
Subject: [PATCH 4/9] Update pcm.c

---
 sound/core/pcm.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/core/pcm.c b/sound/core/pcm.c
index e1e9e0c..8c42d01 100644
--- a/sound/core/pcm.c
+++ b/sound/core/pcm.c
@@ -215,6 +215,10 @@ static char *snd_pcm_format_names[] = {
 	FORMAT(G723_40_1B),
 	FORMAT(DSD_U8),
 	FORMAT(DSD_U16_LE),
+	FORMAT(DSD_U32_LE),
+	FORMAT(DSD_U16_BE),
+	FORMAT(DSD_U32_BE),
+	
 };
 
 const char *snd_pcm_format_name(snd_pcm_format_t format)
-- 
1.7.10.4

